[CN-499] Add Documentation for Restore from HotBackup CR (#59)
hazelcast/hazelcast-platform-operator-docs: docs/modules/ROOT/pages/backup-restore.adoc

=== Restoring from Local Backups

To restore a cluster from local backups, you can directly reapply the `Hazelcast` resource, and it will have access to the PVCs that contain the persisted data. This will restore the Hazelcast cluster from existing `hot-restart` folders.

However, if you want to restore from local backups you have taken using the `HotBackup` resource, you can give the `HotBackup` resource name in the restore configuration. For restore to work correctly, make sure the status of the `HotBackup` resource is `Success`. 

When restore mechanism is used, the Restore Agent container will

> be deployed with the Hazelcast container in the same Pod. The agent will start as an initContainer before the Hazelcast container.

Author: Should we remove `be` here?

=== Restoring from External Backups

To restore a cluster from external backups, you can either set up the bucket configuration or give the `HotBackup` resource name you triggered the external backup. In either case backups will be restored from the external buckets.

When restore mechanism is used, the Restore Agent container will

> be deployed with the Hazelcast container in the same Pod. The agent will start as an initContainer before the Hazelcast container.

Author: Same as above

kubectl delete hazelcast hazelcast
----

. Apply the `Hazelcast` resource with restore configuration:
+
[tabs]
====

Bucket Configuration::
+
--
[source,yaml,subs="attributes+"]
----
include::ROOT:example$/hazelcast-persistence-restore-external.yaml[]
----
<1> The bucket URI where backup data will be restored from.
<2> Name of the secret with credentials for accessing the given Bucket URI.
<3> The agent which is responsible for restoring data from the external storage.
    The agent configuration is optional. If you give `restore` under the `persistence` and do not pass the agent configuration, the operator
    will use the latest agent version compatible with the operator version.
+

. If you have not created the secret, you must create the secret using the same way with the <<create-secret, Triggering External Backups>>.

--

HotBackup resource name::
+
--
[source,yaml,subs="attributes+"]
----
include::ROOT:example$/hazelcast-persistence-restore-cr-name.yaml[]
----

<1> The `HotBackup` resource name that will be used for restore. The bucket URI and secret will be taken from the `HotBackup` resource.
<2> The agent which is responsible for restoring data from the external storage.

>     The agent configuration is optional. If you give `restore` under the `persistence` and do not pass the agent configuration, the operator

Reviewer: ```suggestion
    The agent configuration is optional. If you provide `restore` under `persistence` and do not pass the agent configuration, Hazelcast Platform Operator
```

>     will use the latest agent version compatible with the operator version.

Reviewer: ```suggestion
    uses the latest agent version that is compatible with its version.
```

--

====
[source,yaml,subs="attributes+"]

. Apply the `Hazelcast` resource:
+
[source,bash]
----
kubectl apply -f ./hazelcast-persistence-restore-external.yaml
----

> . See that the same PVCs are bound to the new pods:

Reviewer: ```suggestion
. You can check that the same PVCs are bound to the new pods:
```
